geth/jail/jail.go

```diff
@@ -115,60 +144,26 @@ func (jail *Jail) Parse(chatID string, js string) string {
 // Call executes the `call` function w/i a jail cell context identified by the chatID.
 // Jail cell is clonned before call is executed i.e. all calls execute w/i their own contexts.
 func (jail *Jail) Call(chatID string, path string, args string) string {
-	jail.RLock()
-	cell, ok := jail.cells[chatID]
-	if !ok {
-		jail.RUnlock()
-		return makeError(fmt.Sprintf("Cell[%s] doesn't exist.", chatID))
-	}
-	jail.RUnlock()
-
-	// Due to the new timer assigned we need to clone existing cell to allow
-	// unique cell runtime and eventloop context.
-	cellCopy, err := cell.Copy()
+	jcell, err := jail.GetCell(chatID)
 	if err != nil {
 		return makeError(err.Error())
 	}
 
-	// isolate VM to allow concurrent access
-	vm := cellCopy.CellVM()
-	res, err := vm.Call("call", nil, path, args)
-
-	return makeResult(res.String(), err)
-}
-
-// JailCellVM returns instance of Otto VM (which is persisted w/i jail cell) by chatID
-func (jail *Jail) JailCellVM(chatID string) (*otto.Otto, error) {
-	if jail == nil {
-		return nil, ErrInvalidJail
-	}
-
-	jail.RLock()
-	defer jail.RUnlock()
-
-	cell, ok := jail.cells[chatID]
-	if !ok {
-		return nil, fmt.Errorf("cell[%s] doesn't exist", chatID)
-	}
-
-	return cell.CellVM(), nil
-}
+	res, err := jcell.Call("call", nil, path, args)
 
-// GetCell returns instance of jailed runtime
-func (jail *Jail) GetCell(chatID string) (common.JailCell, error) {
-	if jail == nil {
-		return nil, ErrInvalidJail
-	}
+	jcell.Lock()
+	defer jcell.Unlock()
```

Comment:
Hm. Why are we acquiring lock here, by the way? `jcell.Call` above already does locking inside as well as `lo.Run` below.